Week 1 - preprocessing

koenkoopen opened the pull request:
Hi all,

Here is my PR for the Week 1 assignment on preprocessing.
I've used the hotel reservations dataset and written the data into a table in my catalog.

Next, I created some preprocessing functions in a similar way to those shown in the lectures.
The results (train and test sets) are written into a table in my catalog.
I have written unit tests using pytest, and these can be run using the pytest notebook.

Looking forward to your feedback!

koenkoopen committed:
upload temp preprocessing code

Commit:
Add preprocessing functionalities without unit tests

Commit:
Add try except blocks and extra logging

koenkoopen committed:
fix pre-commit hooks

Commit:
add fillna and some docstrings missing

Commit:
Finished preprocessing code, add raise exceptions after catching and logging

koenkoopen committed:
add initial unit tests

Commit:
Fixed most unit tests

Commit:
Remove some code to fix linters

Commit:
remove asserts

koenkoopen committed:
fix ruff linter errors

Commit:
Added final tests

basakeskili commented on pyproject.toml:
```diff
@@ -33,10 +35,12 @@ build-backend = "setuptools.build_meta"
 
 [tool.setuptools.packages.find]
 where =["src"]
-include = ["house_price*"]
```
hey Koen, it looks good but I have a few suggestions:

- rename src to house prices or create a new folder under src with the name house_pice
instead of this _from src.utils import configure_logging_
you use from _house_price.utils import configure_logging_

koenkoopen replied:
Thanks for noticing, I've overlooked this. It's still the old configuration of the template

koenkoopen replied:
Done

basakeskili commented on pyproject.toml:
```diff
@@ -1,5 +1,5 @@
 [project]
-name = "housing-price"
+name = "koen_databricks_mlops"
```
project name is the distribution name of your package - it is the name we use to install packages. Of course it's not needed for this course as we won't publish them to pip but still a good practice to use names consistent :)

for example:
scikit-learn (distribution name, so we use _pip install scikit-learn_) 
but we use _import sklearn_

koenkoopen replied:
Done

koenkoopen committed:
Update pyproject.toml, rename src to hotel_reservation and project name to hotel-reservation

Update pyproject.toml, rename src to hotel_reservation and project name to hotel-reservation

koenkoopen committed:
rename src to hotel_reservation

basakeskili reviewed (approved):
Thanks!
